### PR TITLE
vieb: use p7zip instead precompiled 7z binary

### LINKED ISSUE #133077: Unable to install extensions on Vieb
### Describe the bug
The Vieb web browser is supposed to have ([limited](https://github.com/Jelmerro/Vieb/issues/130)) support for installing Chrome extensions (such as Dark Reader). When I try to do this, it gives me an `ENOENT` error.

### Steps To Reproduce
1. Install Vieb via nixpkgs.
2. `vieb 'https://chrome.google.com/webstore/detail/dark-reader/eimadpbcbfnmbkopoojfekhnkhdbieeh'` to open Dark Reader extension.
3. Type `:extensions install` to try to install extension.

This leads to an `ENOENT` error.

### Expected behavior
It should install the extension.

### Notify maintainers
@gebner @fortuneteller2k 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.52, NixOS, 21.05.1669.973910f5c31 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.14`
 - channels(root): `"nixos-21.05.1669.973910f5c31, nixpkgs-unstable-21.11pre304227.ecaf3da9340"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by j-hui**
###### Motivation for this change

Fix #133077 .

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by j-hui** [comment]
Currently, the `libexec/vieb/node_modules/7zip-bin/linux/x64/7za` seems to be succesfully patched, but Vieb is still unable to install extensions with this change.

**Comment by symphorien** [comment]
how about replacing the faulty 7z-bin by nixpkgs's one ?

**Comment by j-hui** [comment]
@symphorien that could also work, but I'm not really sure how to do that. Do you have any pointers for how to do that?

I'm hesitant to just copy in nixpkgs's 7zip executable into the `node_modules/7zip-bin` directory since the `7z` binary is pulled in as part of the `7zip-bin` yarn package, and I'm worried that copying in the executable is fragile.

**Comment by symphorien** [comment]
> I'm hesitant to just copy in nixpkgs's 7zip executable into the node_modules/7zip-bin directory 

that's exactly what I was thinking about. A symlink would be even better. Can you elaborate how this would be fragile?

**Comment by j-hui** [comment]
I'm worried it's fragile because it seems that I would need to accont for different archs/platforms, and I'm not sure how to test that. 

Using the autopatchelf script abstracts away from arch/platform-specific paths. What's the downside to this approach?

In any case, I tried your symlinking suggestion (something like `ln -s -f ${p7zip}/bin/7z $out/libexec/vieb/node_modules/7zip-bin/linux/x64/7za`) and the resulting 7z binary works fine (on linux/x64). But Vieb is still unable to install extensions (there's still something wrong with it that I need to fix).

**Comment by symphorien** [comment]
Well the thing is, patchelfHook is not the most robust solution either. For example if the node package starts shipping binaries compiled with more recent glibc version that nixos, then the executable will fail at runtime.
Regarding the architecture, anyway the 7z executable from nixpkgs has the *right* architecture, so even if you replace *all* executable files named `7z*` by a symlink to the store with `find`, this will be correct.
In the end, as de facto you are maintaining this package, I'll leave the decision to you, but I think using 7z from nixpkgs seems better to me.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133151` at 2a6c2b4a run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>vieb</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

yarn is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:294:7:
```
    |
294 |       buildInputs = [ yarn nodejs rsync ] ++ extraBuildInputs;
    |       ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

rsync is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:294:7:
```
    |
294 |       buildInputs = [ yarn nodejs rsync ] ++ extraBuildInputs;
    |       ^
```
</li>
  </ul>
</details>

**Comment by gebner** [comment]
We're already replacing electron with the nixpkgs-built version.  I'd prefer to use p7zip from nixpkgs as well.  And please replace all `7za` binaries, these are taking up ten useless megabytes.

Either way, I think we should wait with merging this PR until the extension installation works.

**Comment by j-hui** [comment]
Got it @symphorien , I updated the PR to do what you suggested, thanks for the feedback!

@gebner I'll wait for that update. Is there a PR or issue associated with that?

File: pkgs/applications/networking/browsers/vieb/default.nix [hunk 1]
```diff
@@ -34,6 +34,11 @@ mkYarnPackage rec {
   };
 
   postInstall = ''
+    unlink $out/libexec/vieb/deps/vieb/node_modules
+    ln -s $out/libexec/vieb/node_modules $out/libexec/vieb/deps/vieb/node_modules
+
+    find $out/libexec/vieb/node_modules/7zip-bin -name 7za -exec ln -s -f ${p7zip}/bin/7z {} ';'
```
Comment (symphorien) [inline pkgs/applications/networking/browsers/vieb/default.nix:40]: It looks like 7za and 7z are not the same?
```suggestion
    find $out/libexec/vieb/node_modules/7zip-bin -name 7za -exec ln -s -f ${p7zip}/bin/7za {} ';'
```

Comment (j-hui) [reply pkgs/applications/networking/browsers/vieb/default.nix]: Oops, sorry about that. Just fixed it.

**Comment by j-hui** [comment]
@gebner , any updates on the electron update that this is blocked on?

**Comment by gebner** [comment]
I don't think this PR requires an electron update.  I just preferred to wait until the extension installation works in vieb.  But let's just merge this PR, it's a clear improvement over the current state.

**Comment by j-hui** [comment]
Got it, thanks @gebner !

**Pull request merged**